src/librustc_interface/passes.rs

```diff
@@ -595,7 +596,22 @@ fn write_out_deps(
         // prevents `make` from spitting out an error if a file is later
         // deleted. For more info see #28735
         for path in files {
+            //FIXME: async!
+            let mut reader = io::BufReader::new(fs::File::open(&path)?);
+            let mut hasher = rustc_data_structures::fx::FxHasher::default();
+            let mut buffer = [0; 1024];
+            loop {
+                let count = reader.read(&mut buffer)?;
+                if count == 0 {
+                    break;
+                }
+                hasher.write(&buffer[..count]);
+            }
+            let hash = hasher.finish();
+            let size = std::fs::metadata(&path)?.len();
```

Comment:
You could avoid this metadata call by counting the amount of read bytes.

Reply:
Cute!